src/neo4j_graphrag/types.py

```diff
@@ -137,11 +139,53 @@ class VectorCypherSearchModel(VectorSearchModel):
     query_params: Optional[dict[str, Any]] = None
 
 
+class HybridSearchRanker(Enum):
+    """Enumerator of Hybrid search rankers."""
+
+    NAIVE = "naive"
+    LINEAR = "linear"
+
+
 class HybridSearchModel(BaseModel):
     query_text: str
     query_vector: Optional[list[float]] = None
     top_k: PositiveInt = 5
     effective_search_ratio: PositiveInt = 1
+    ranker: Union[str, HybridSearchRanker] = HybridSearchRanker.NAIVE
+    alpha: Optional[float] = None
+
+    @field_validator("ranker", mode="before")
+    def validate_ranker(cls, v: Union[str, HybridSearchRanker]) -> HybridSearchRanker:
+        if isinstance(v, str):
+            try:
+                return HybridSearchRanker(v.lower())
+            except ValueError:
+                allowed = ", ".join([r.value for r in HybridSearchRanker])
+                raise ValueError(
+                    f"Invalid ranker value. Allowed values are: {allowed}."
+                )
+        elif isinstance(v, HybridSearchRanker):
+            return v
+        else:
+            allowed = ", ".join([r.value for r in HybridSearchRanker])
+            raise ValueError(f"Invalid ranker type. Allowed values are: {allowed}.")
+
+    @model_validator(mode="after")
+    def validate_alpha(self) -> Self:
+        ranker, alpha = self.ranker, self.alpha
+        if ranker == HybridSearchRanker.LINEAR:
+            if alpha is None:
+                raise ValueError("alpha must be provided when using the linear ranker")
+            if isinstance(alpha, float) and not (0.0 <= alpha <= 1.0):
```

Comment:
Is the `isinstance` test needed? Isn't this validator run after Pydantic "normal" validation, that includes the type check?

Reply:
Previously a mypy test failed but this works now https://github.com/neo4j/neo4j-graphrag-python/pull/284/commits/bb4d2f9e707e1725b96bcafc8f413e88a2b4e36a